File: demo/src/main/java/com/example/demo/service/RatingService.java
package com.example.demo.service;

import com.example.demo.dto.AverageRatingResponse;
import com.example.demo.dto.RatingRequest;
import com.example.demo.dto.RatingResponse;
import com.example.demo.dto.Response;

import java.util.List;

public interface RatingService {
    
    // Add rating
    Response<RatingResponse> addRating(RatingRequest request);
    
    // Get all ratings for a food
    Response<List<RatingResponse>> getAllRatingsForFood(Long foodId);
    
    // Get average rating and total count for a food (Public)
    Response<AverageRatingResponse> getAverageRatingResponse(Long foodId);
    
    // Get all ratings (Admin)
    Response<List<RatingResponse>> getAllRatings();

    // Delete rating by ID
    Response<Void> deleteRating(Long ratingId);

    // Get average rating for a food
    Double getAverageRating(Long foodId);

    // Get total ratings count for a food
    Long getTotalRatings(Long foodId);
}

File: food-frontend/my-food-app/src/services/rating.service.js
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8080/api';

export const addRating = async (data) => {
    try {
        const response = await axios.post(`${API_BASE_URL}/ratings`, data);
        return response.data;
    } catch (e) {
        console.error('Error adding rating:', e);
        throw e;
    }
};

export const getAllRatingsForFood = async (foodId) => {
    try {
        const response = await axios.get(`${API_BASE_URL}/ratings/food/${foodId}`);
        return response.data;
    } catch (e) {
        console.error('Error fetching ratings for food:', e);
        throw e;
    }
};

export const getFoodAverageRating = async (foodId) => {
    try {
        const response = await axios.get(`${API_BASE_URL}/ratings/food/${foodId}/average`);
        return response.data;
    } catch (e) {
        console.error('Error fetching average rating:', e);
        throw e;
    }
};

export const deleteRating = async (ratingId) => {
    try {
        const response = await axios.delete(`${API_BASE_URL}/ratings/${ratingId}`);
        return response.data;
    } catch (e) {
        console.error('Error deleting rating:', e);
        throw e;
    }
};

export const getAllRatings = async () => {
    try {
        const response = await axios.get(`${API_BASE_URL}/ratings`);
        return response.data;
    } catch (e) {
        console.error('Error fetching all ratings:', e);
        throw e;
    }
};
